library/cohesity_job.py

```diff
@@ -778,6 +778,34 @@ def update_vmware_job(module, job_meta_data, job_details):
         )
 
 
+def delete_sources(module, job_meta_data, job_details):
+    job_details['environment'] = 'Physical'
+    for source in module.params.get('protection_sources'):
+        job_details['endpoint'] = source['endpoint']
+        source_id = get__prot_source_id__by_endpoint(
+            module, job_details)
+        if source_id in job_meta_data['sourceIds']:
+            job_meta_data['sourceIds'].remove(source_id)
+        else:
+            continue
+        if module.params.get('environment') == 'PhysicalFiles':
+            for source in job_meta_data['sourceSpecialParameters']:
+                if source['sourceId'] == source_id:
+                    index = job_meta_data['sourceSpecialParameters'].index(source)
+                    del job_meta_data['sourceSpecialParameters'][index]
+    if len(job_meta_data['sourceIds']) == 0:
```

Comment:
probably a good idea to add what failed.. say they are trying to remove 10 sources .. it partially failed. then it'd be useful. Else if its all or none update, then clearly explain to user why it failed.